ckanext/dcatapchharvest/profiles.py

```diff
@@ -269,34 +275,41 @@ def _temporals(self, subject, predicate):
 
         return temporals
 
-    def _clean_datetime(self, datetime_value):
+    def _clean_datetime(self, datetime_value, data_type):
         """Convert a literal in one of the accepted data types into an isoformat
         datetime string.
 
         Accepted types are: xsd:date, xsd:dateTime, xsd:gYear, or
         xsd:gYearMonth; or schema:Date or schema:DateTime, for temporals
         specified as schema:startDate and schema:endDate.
+
+        We only consider the parts of the date that are expected from the given
+        data_type, e.g. the year of an xsd:gYear, even if the month and day
+        have been included in the datetime_value. If a datetime_value with
+        data_type of xsd:dateTime or schema:DateTime does not contain time
+        information, we discard it.
```

Comment:
awesome, I think this is already better than what I imagined for the readme. I think a pointer to "what is generally going on with dates when harvesting" would be worthwhile, though, if it doesn't already exist.

Reply:
Good point, I will add it.

Reply:
Awesome! great documentation, thank you very much :)